core/block/template/service.go

```diff
@@ -79,32 +80,60 @@ func (s *service) Init(a *app.App) error {
 	return nil
 }
 
-// StateFromTemplate creates clone of template object state with empty localDetails and updated objectTypes.
+// CreateTemplateStatePreservingDetails creates clone of template object state with empty localDetails and updated objectTypes.
 // Blank template is created in case template object is deleted or blank/empty templateIв is provided
-func (s *service) StateFromTemplate(templateId, name string) (st *state.State, err error) {
+func (s *service) CreateTemplateStateWithDetails(
+	templateId string,
+	details *types.Struct,
+) (targetState *state.State, err error) {
 	if templateId == BlankTemplateId || templateId == "" {
-		return s.blankTemplateState(), nil
+		targetState = s.createBlankTemplateState()
+	} else {
+		targetState, err = s.createCustomTemplateState(templateId)
+		if err != nil {
+			return
+		}
 	}
-	if err = getblock.Do(s.picker, templateId, func(b smartblock.SmartBlock) (innerErr error) {
-		if lo.Contains(b.ObjectTypeKeys(), bundle.TypeKeyTemplate) {
-			st, innerErr = s.getNewPageState(b, name)
+
+	targetState = targetState.NewState()
+	targetState.AddDetails(details)
+
+	return targetState, nil
+}
+
+func (s *service) createCustomTemplateState(templateId string) (targetState *state.State, err error) {
+	err = getblock.Do(s.picker, templateId, func(sb smartblock.SmartBlock) (innerErr error) {
+		if lo.Contains(sb.ObjectTypeKeys(), bundle.TypeKeyTemplate) {
```

Comment:
We can invert if here to reduce complexity
`if !lo.Contains() {
return fmt.Errorf()
}`